frontend/LatexRenderer.tsx

```
@@ -0,0 +1,79 @@
+        if (isLatex) {
+          try {
+            const html = katex.renderToString(latexContent, {
+              throwOnError: false, // Don't crash the app on bad LaTeX
+              displayMode: displayMode,
+              output: 'htmlAndMathml', // Good for accessibility
+              // You can add macros here if needed:
+              // macros: {"\\RR": "\\mathbb{R}"}
+            });
+            // KaTeX generates HTML, so we need to render it dangerously.
+            // This is generally safe as KaTeX output is controlled.
+            return <span key={index} dangerouslySetInnerHTML={{ __html: html }} />;
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)